tests/filtertransformers/test_mongo.py

```diff
@@ -341,8 +342,170 @@ def test_not_implemented(self):
                 'HAS ANY > 3:"He":>55.3 , = 6:>"Ti":<37.6 , 8:<"Ga":0'
             )
 
-        with self.assertRaises(VisitError):
-            self.transform("list LENGTH > 3")
+        self.assertEqual(
+            self.transform("list LENGTH > 3"), {"list.4": {"$exists": True}}
+        )
+
+    def test_list_length_aliases(self):
+        from optimade.server.mappers import StructureMapper
+
+        transformer = MongoTransformer(mapper=StructureMapper())
+        parser = LarkParser(version=self.version, variant=self.variant)
+
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH 3")), {"nelements": 3}
+        )
+
+        self.assertEqual(
+            transformer.transform(
+                parser.parse('elements HAS "Li" AND elements LENGTH = 3')
+            ),
+            {"$and": [{"elements": {"$in": ["Li"]}}, {"nelements": 3}]},
+        )
+
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH > 3")),
+            {"nelements": {"$gt": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH < 3")),
+            {"nelements": {"$lt": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("elements LENGTH = 3")), {"nelements": 3}
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH <= 3")),
+            {"nsites": {"$lte": 3}},
+        )
+        self.assertEqual(
+            transformer.transform(parser.parse("cartesian_site_positions LENGTH >= 3")),
+            {"nsites": {"$gte": 3}},
+        )
+
+    def test_unaliased_length_operator(self):
+        transformer = MongoTransformer()
+        parser = LarkParser(version=self.version, variant=self.variant)
```

Comment:
Same here as the previous comment. And even the transformer doesn't need to be redefined, since it should be `t` from line 16?

Reply:
The transformer in these tests needs to take a mapper as an argument, so the same one can't be reused. This means the parser also can't be reused (its out of scope in the methods, we only expose the transform method which is tied to the transformer).

Reply:
If you see the `def setUp()`, which runs just prior of this test method, why wouldn't that suffice for the tests in this method?

Reply:
> The transformer in these tests needs to take a mapper as an argument (...)

But in this exact test method, the transformer _doesn't_ take a mapper as an argument? I am confused :( I see that it makes sense to instantiate a new transformer for the other test methods you added, but not for this one.

And how is the parser out of scope? It doesn't depend on the transformer as far as I know? Or? :|

Reply:
> The transformer in these tests needs to take a mapper as an argument, so the same one can't be reused. This means the parser also can't be reused (its out of scope in the methods, we only expose the transform method which is tied to the transformer).

AAAHHH I see the whole problem now - sorry!! Yeah, we set both the parser and transformer in the `setUp` method, but only keep the `transform` in the instance..... right, right, right

Reply:
Replied to the wrong comment initially, sorry! I've amended my commit with your suggestion

Reply:
> So would it make sense then to expose the instantiated transformer and parser?

The other two new tests can't use this anyway, so I think it's now fine as is

Reply:
Cool. I'll stop poking at this.

(although, just tested locally changing this to `setUpClass()`, and it works only if one then makes `transform` an actual separate class method, and consequently "saves" the instantiated parser and transformer as class attributes.